File: backend/src/main/java/com/example/aibackend/dto/UserDTO.java
package com.example.aibackend.dto;

import java.time.LocalDate;

public class UserDTO {
    private Long id;
    private String firstName;
    private String lastName;
    private String login;
    private LocalDate birthDate;
    private String address;

    public UserDTO() {}

    public UserDTO(Long id, String firstName, String lastName, String login, LocalDate birthDate, String address) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.login = login;
        this.birthDate = birthDate;
        this.address = address;
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getFirstName() { return firstName; }
    public void setFirstName(String firstName) { this.firstName = firstName; }
    public String getLastName() { return lastName; }
    public void setLastName(String lastName) { this.lastName = lastName; }
    public String getLogin() { return login; }
    public void setLogin(String login) { this.login = login; }
    public LocalDate getBirthDate() { return birthDate; }
    public void setBirthDate(LocalDate birthDate) { this.birthDate = birthDate; }
    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }
}

File: android/app/src/main/java/ch/goodone/angularai/android/data/remote/dto/UserDTO.kt
package ch.goodone.angularai.android.data.remote.dto

data class UserDTO(
    val id: Long? = null,
    val firstName: String? = null,
    val lastName: String? = null,
    val login: String? = null,
    val email: String? = null,
    val birthDate: String? = null, // yyyy-MM-dd
    val address: String? = null,
    val role: String? = null,
    val password: String? = null
)
